### DISCUSSION TITLE
how can I build statically linked cardano node binaries?

### DISCUSSION THREAD

**Discussion by monkey-jsun** [IntersectMBO/cardano-node#4677] [General]
I am following the guide below to build cardano node binaries on ubuntu 20.04.  At the end I got a set of dynamically linked executables.  How do I build statically linked executables (like cardano-wallet project does)?

https://developers.cardano.org/docs/get-started/installing-cardano-node/

**Comment by angerman** [comment]
Static binaries for which architecture? x86_64 or AArch64 or something else?

https://github.com/input-output-hk/tools 

Has nix expressions for fully static (muslc based) builds. For example this one: [cardano-node.x86_64-linux-musl](https://ci.zw3rk.com/build/942525)

You can build them yourself with

```
nix build -f release cardano-node.x86_64-linux-musl
```
From that repo.

**Reply by monkey-jsun** [reply]
It is for x86_64.  

Thanks for your reply.  Not sure if I understand.  Are you saying I cannot build statically linked binaries by using below process and I have to use a different process (i.e., different C library)?

https://developers.cardano.org/docs/get-started/installing-cardano-node/

**Reply by angerman** [reply]
You can follow the general process but you will need to link against a statically linkable libc (e.g. muslc), as well as ensure that all you other dependencies are built against it (including the Haskell compiler, producing code for that libc target). You'll also want to ensure that you do not end up having shared libraries in your library search path. That would end up producing dynamically linked executables.
